src/methods/iterator.js

```diff
@@ -177,12 +269,18 @@ export function IteratorClose(realm: Realm, iterator: ObjectValue, completion: A
   if (ret instanceof UndefinedValue) return completion;
 
   // 5. Let innerResult be Call(return, iterator, « »).
-  let innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  let innerResult;
+  try {
+    innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  } catch (error) {
+    innerResult = error;
+  }
```

Comment:
You DO have to catch and cache regular AbruptCompletions, but this catch also swallows other kinds of errors happen because of internal errors in the interpreter (i.e. !(error instanceof AbruptCompletion)) or because the source code contains a construct that Prepack cannot deal with when abstract values are involved (i.e. error instanceof IntrospectionThrowCompletion).